testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/forms/LogoutTest.java

```diff
@@ -118,6 +134,37 @@ public void logoutSession() {
         assertNotEquals(sessionId, sessionId2);
     }
 
+    @Test
+    public void logoutWithExpiredSession() throws Exception {
+        try (AutoCloseable c = new RealmAttributeUpdater(adminClient.realm("test"))
+                .updateWith(r -> r.setSsoSessionMaxLifespan(2))
+                .update()) {
+
+            oauth.doLogin("test-user@localhost", "password");
+
+            String code = oauth.getCurrentQuery().get(OAuth2Constants.CODE);
+
+            oauth.clientSessionState("client-session");
+            OAuthClient.AccessTokenResponse tokenResponse = oauth.doAccessTokenRequest(code, "password");
+            String idTokenString = tokenResponse.getIdToken();
+
+            // wait for a timeout
+            // setTimeOffset doesn't work because session cookie is not invalidated thus the logout flow would continue with browser logout
+            TimeUnit.SECONDS.sleep(3);
+
+            String logoutUrl = oauth.getLogoutUrl().redirectUri(oauth.APP_AUTH_ROOT).idTokenHint(idTokenString).build();
+            driver.navigate().to(logoutUrl);
+
+            // should not throw an internal server error
+            assertTrue(appPage.isCurrent());
```

Comment:
Nit: should be 
```suggestion
            appPage.assertCurrent();
```
to have nicer failure report.